lib/Generate.js

```diff
@@ -34,6 +34,22 @@ function prepareCompiledJsFile(pipeFilename /*: string */, dest /*: string */) {
   });
 }
 
+const testVariantDefinition = /^var\s+\$elm_explorations\$test\$Test\$Internal\$(UnitTest|FuzzTest|Labeled|Skipped|Only|Batch)\s*=\s*(?:\w+\(\s*)?function\s*\([\w, ]*\)\s*\{\s*return\s*\{\s*\$:\s*(['"])\1\2/gm;
+
+const checkDefinition = /^(var\s+\$author\$project\$Test\$Runner\$Node\$check\s*=\s*function\s*\(value\)\s*\{\s*)return\s+(\$elm\$core\$Maybe\$)Nothing;?(\s*\};?)/m;
+
+function hackCompiledElmJs(content) {
+  return (
+    'var __elmTestSymbol = Symbol("elmTestSymbol");\n' +
+    content
+      .replace(testVariantDefinition, '$&, __elmTestSymbol: __elmTestSymbol')
+      .replace(
+        checkDefinition,
+        '$1return value && value.__elmTestSymbol === __elmTestSymbol ? $2Just(value) : $2Nothing;$3'
+      )
+  );
+}
```

Comment:
Here are the JS hacks relying on the JS output of Elm. Also note that if the `Test` type every gets more variants, we need to add them to the first regex.

Reply:
Could we consider instead changing the names of the custom type constructors (the definition is [here](https://github.com/elm-explorations/test/blob/b639def0fdb3437cd076d4ed840cb2451c553f7c/src/Test/Internal.elm#L10)) like so:

```elm


type Test
    = ElmTestVariant__UnitTest (() -> List Expectation)
    | ElmTestVariant__FuzzTest (Random.Seed -> Int -> List Expectation)
    | ElmTestVariant__Labeled String Test
    | ElmTestVariant__Skipped Test
    | ElmTestVariant__Only Test
    | ElmTestVariant__Batch (List Test)
```

and then we could use the regex to check for the `ElmTestVariant__` bit. That way it should be obvious to future contributors to the test library that they need to add an `ElmTestVariant__` and the test runner would immediately pick it up?

Reply:
That sounds like a good idea! I can make a PR to elm-explorations/test as well. We can support both the old and new names in the regex so the runner works with both old and new versions of the test framework.

Reply:
Brilliant!

Reply:
https://github.com/elm-explorations/test/pull/153